test/performance/sensors_system.cc

```diff
@@ -0,0 +1,135 @@
+#include "test_config.hh"
+
+#include "../helpers/EnvTestFixture.hh"
+
+/////////////////////////////////////////////////
+void getMemInfo(double &_resident, double &_share)
+{
+#ifdef __linux__
+  int totalSize, residentPages, sharePages;
+  totalSize = residentPages = sharePages = 0;
+
+  std::ifstream buffer("/proc/self/statm");
+  buffer >> totalSize >> residentPages >> sharePages;
+  buffer.close();
+
+  // in case x86-64 is configured to use 2MB pages
+  int64_t pageSizeKb = sysconf(_SC_PAGE_SIZE) / 1024;
+
+  _resident = residentPages * pageSizeKb;
+  _share = sharePages * pageSizeKb;
+#elif __MACH__
+  // /proc is only available on Linux
+  // for OSX, use task_info to get resident and virtual memory
+  struct task_basic_info t_info;
+  mach_msg_type_number_t t_info_count = TASK_BASIC_INFO_COUNT;
+  if (KERN_SUCCESS != task_info(mach_task_self(),
+                                TASK_BASIC_INFO,
+                                (task_info_t)&t_info,
+                                &t_info_count))
+  {
+    gzerr << "failure calling task_info\n";
+    return;
+  }
+  _resident = static_cast<double>(t_info.resident_size/1024);
+  _share = static_cast<double>(t_info.virtual_size/1024);
+#else
+  gzerr << "Unsupported architecture\n";
+  return;
+#endif
+}
+
+/////////////////////////////////////////////////
+class SensorsSystemFixture : public InternalFixture<::testing::Test>
+{
+};
+
+
+/////////////////////////////////////////////////
+// Test repeatedly launching sim with sensors system and check for
+// memory leaks
+TEST_F(SensorsSystemFixture, GZ_UTILS_TEST_DISABLED_ON_MAC(MemLeak))
+{
+  // max memory change allowed
+  const double resMaxPercentChange = 1.0;
+  const double shareMaxPercentChange = 1.0;
+
+  std::vector<double> residentMemV;
+  std::vector<double> shareMemV;
+
+  for (unsigned int i = 0; i < 5; ++i)
+  {
+    // get resident and shared memory usage
+    double residentMem = 0;
+    double shareMem = 0;
+    getMemInfo(residentMem, shareMem);
+
+    residentMemV.push_back(residentMem);
+    shareMemV.push_back(shareMem);
+
+    gz::sim::ServerConfig serverConfig;
+
+    const std::string sdfFile = std::string(PROJECT_SOURCE_PATH) +
+        "/test/worlds/sensors_system_mem_leak.sdf";
+
+    serverConfig.SetSdfFile(sdfFile);
+
+    gz::sim::Server server(serverConfig);
+    server.Run(true, 100, false);
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+  }
+
+  // Calculate the percent change between runs
+  // Start from index 1 (instead of 0) since there is a big jump in memory usage
+  // when the system is first loaded. Make sure subsequent runs do not continue
+  // to increase memory usage
+  for (unsigned int i = 1; i < residentMemV.size() - 1; ++i)
+  {
+    double resPercentChange =
+        (residentMemV[i+1] - residentMemV[i]) / residentMemV[i];
+    double sharePercentChange = (shareMemV[i+1] - shareMemV[i]) / shareMemV[i];
+
+    gzdbg << "ResPercentChange[" << resPercentChange << "] "
```

Comment:
Since you aren't explicitly setting the console log level, I don't know that these are guaranteed to come out as part of the test results?

Reply:
added a call to set verbosity level to make sure we see these console messages [bd3ff2e](https://github.com/gazebosim/gz-sim/pull/1960/commits/bd3ff2eb5491808362623851ffcde138b07229e7)